app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -85,46 +95,67 @@ class HomeActivityTest {
     Intents.release()
   }
 
-  @Test
-  fun testHomeActivity_firstOpen_hasWelcomeString() {
-    launch(HomeActivity::class.java).use {
-      onView(
-        atPositionOnView(
-          R.id.home_recycler_view,
-          0,
-          R.id.welcome_text_view
-        )
-      ).check(matches(withText("Welcome to Oppia!")))
-    }
+  private fun setUpTestApplicationComponent() {
+    DaggerHomeActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
   }
 
   @Test
-  fun testHomeActivity_secondOpen_hasWelcomeBackString() {
-    simulateAppAlreadyOpened()
-
+  fun testHomeActivity_recyclerViewIndex0_displayProfileName_profileNameDisplayedSuccessfully() {
     launch(HomeActivity::class.java).use {
-      // Wait until the expected text appears on the screen, and ensure it's for the welcome text view.
-      waitForTheView(withText("Welcome back to Oppia!"))
       onView(
         atPositionOnView(
           R.id.home_recycler_view,
           0,
-          R.id.welcome_text_view
+          R.id.profile_name_textview
         )
-      ).check(matches(withText("Welcome back to Oppia!")))
+      ).check(matches(withText("Sean!")))
     }
   }
 
   @Test
-  fun testHomeActivity_recyclerViewIndex0_displaysWelcomeMessageCorrectly() {
+  fun testHomeActivity_recyclerViewIndex0_displayGreetingMessageBasedOnTime_greetingMessageDisplayedSuccessful() {
     launch(HomeActivity::class.java).use {
-      onView(
-        atPositionOnView(
-          R.id.home_recycler_view,
-          0,
-          R.id.welcome_text_view
-        )
-      ).check(matches(withText(containsString("Welcome"))))
+      val c = Calendar.getInstance()
+      when (c.get(Calendar.HOUR_OF_DAY)) {
+        in 5..11 -> {
+          onView(
+            atPositionOnView(
+              R.id.home_recycler_view,
+              0,
+              R.id.welcome_text_view
+            )
+          ).check(matches(withText("Good morning,")))
+        }
+        in 12..16 -> {
+          onView(
+            atPositionOnView(
+              R.id.home_recycler_view,
+              0,
+              R.id.welcome_text_view
+            )
+          ).check(matches(withText("Good afternoon,")))
+        }
+        in 17..22 -> {
+          onView(
+            atPositionOnView(
+              R.id.home_recycler_view,
+              0,
+              R.id.welcome_text_view
+            )
+          ).check(matches(withText("Good evening,")))
+        }
+
+        else -> onView(
```

Comment:
Update this test cases as per implementation change update

Reply:
Updated.